### PR TITLE
Feature/124 update application page

### PR DESCRIPTION
**Pull request by RohitGarudadri**
### Tickets:

Fixes # (129, 130)

### List of changes:

Updated validation schema for applications and added another essay question and changed some field names and edited the hacker model to be grouped together to be consistent with backend.

### Type of change:

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### How did you do this?

By going through the ManageApplicationContainer file and editing the appropriate fields and the validationSchema for the Applications and editing the IHacker model in userTypes to be grouped together.

### Why are you choosing this approach?

Its the only approach to solve this.

### Questions for code reviewers?

Could you make sure I didn't leave any random redundant/useless statements anywhere?

### PR Checklist:

- [x] Merged `develop` branch (before testing)
- [x] Linted my code locally
- [x] Tested all links in project relevant browsers
- [x] Tested all links on different screen sizes
- [x] Referenced all useful info (issues, tasks, etc)

### CODE REVIEW

File: src/config/userTypes.ts [hunk 1]
```diff
@@ -32,33 +32,43 @@ export interface IHacker {
...
-  needsBus?: boolean;
   application: {
-    portfolioURL: {
-      // gcloud bucket link
-      resume: string;
-      github?: string;
-      dropler?: string;
-      personal?: string;
-      linkedIn?: string;
-      other?: string;
```
Comment (pierreTklein) [inline src/config/userTypes.ts]: This schema follows the mongo schema in the database, and cannot be changed on a whim like this unfortunately

**Comment by RohitGarudadri** [comment]
Yes, this change was done simultaneously along #603, changes were done both in the front and backend schemas to match.

On Dec 4, 2019, at 08:37, Pierre Theo Klein <notifications@github.com<mailto:notifications@github.com>> wrote:


@pierreTklein commented on this pull request.

Is there a corresponding change on the api-end?

________________________________

In src/config/userTypes.ts<https://github.com/hackmcgill/dashboard/pull/733#discussion_r353746487>:

>    application: {
-    portfolioURL: {
-      // gcloud bucket link
-      resume: string;
-      github?: string;
-      dropler?: string;
-      personal?: string;
-      linkedIn?: string;
-      other?: string;


This schema follows the mongo schema in the database, and cannot be changed on a whim like this unfortunately

File: src/config/userTypes.ts (hunk 1 above)
Comment (loreina) [reply src/config/userTypes.ts]: https://github.com/hackmcgill/hackerAPI/pull/603

File: src/features/SingleHacker/SingleHackerView.tsx [hunk 2]
```diff
@@ -149,21 +149,30 @@ class SingleHackerView extends React.Component<
...
+                label="Degree"
+                text={hacker.application.general.degree}
+              />
               <SHField label="Status" text={hacker.status} />
-              <SHField label="Graduation Year" text={hacker.graduationYear} />
+              <SHField
+                label="Graduation Year"
+                text={hacker.application.general.graduationYear}
+              />
               <SHField label="Major(s)" text={hacker.major} />
```
Comment (loreina) [inline src/features/SingleHacker/SingleHackerView.tsx]: should be `fieldOfStudy`

File: src/config/userTypes.ts [hunk 3]
```diff
@@ -32,33 +32,43 @@ export interface IHacker {
...
+      question2: string;
+    };
+    other: {
+      // no enum for these
+      ethnicity: string[];
+      gender: string;
+      codeOfConduct_MLH: boolean;
+      codeOfConduct_MCHACKS: boolean;
+    };
+    accomodation?: {
```
Comment (loreina) [inline src/config/userTypes.ts]: should be spelled `accommodation`

accommodation section is also missing travel, shirt size, impairments, barriers

File: src/config/userTypes.ts [hunk 4]
```diff
@@ -32,33 +32,43 @@ export interface IHacker {
...
+      // "Why do you want to come to our hackathon?"
+      question1: string;
+      // "Some q"
+      question2: string;
+    };
+    other: {
+      // no enum for these
+      ethnicity: string[];
+      gender: string;
+      codeOfConduct_MLH: boolean;
```
Comment (loreina) [inline src/config/userTypes.ts]: one should be `codeOfConduct` and one should be `privacyPolicy`

File: src/config/userTypes.ts [hunk 5]
```diff
@@ -32,33 +32,43 @@ export interface IHacker {
...
+      comments?: string;
+      // "Why do you want to come to our hackathon?"
+      question1: string;
+      // "Some q"
+      question2: string;
+    };
+    other: {
+      // no enum for these
+      ethnicity: string[];
+      gender: string;
```
Comment (loreina) [inline src/config/userTypes.ts]: `gender` is moved to profile now

File: src/config/userTypes.ts (hunk 3 above)
Comment (RohitGarudadri) [reply src/config/userTypes.ts]: Those fields are part of the Account, and not the hacker though. Should I still have them here?

File: src/config/userTypes.ts (hunk 5 above)
Comment (RohitGarudadri) [reply src/config/userTypes.ts]: There is no section for profile, should I create it? Or should it be part of the Account userType instead?

Comment (loreina) [reply src/config/userTypes.ts]: profile = account creation, @logan-r added it in his pr

File: src/config/userTypes.ts (hunk 3 above)
Comment (loreina) [reply src/config/userTypes.ts]: yup move them to hacker app

**Pull request merged**